docs/UserGuide.md

```diff
@@ -266,69 +262,95 @@ Example:
+
+WorkBook adds a new state for an <em>undoable</em> command and uses a list to keep track of all its previous states, allowing the `undo` command to be
+executed as many times to restore the WorkBook to any of its previous tracked state. 
 
 Format: `undo`
+* Undo the previous <em>undoable</em> command.
+* You can execute the `undo` command as many times as the number of states in the Workbook's list 
+of tracked states i.e. stacking 2 undo commands sequentially will revert the WorkBook back to its state 2
+states ago, assuming the WorkBook has executed at least 2 <em>undoable</em> commands previously.
+* If you have not executed an <em>undoable</em> command previously, the WorkBook will remain in its current state 
+and return an error message: "No previous changes to undo!".
+* <em>Undoable</em> commands are stacked, when an <em>undoable</em> command is executed, it will be stacked on top of the
+previous <em>undoable</em> command in the stack. The `undo` command will undo the first <em>undoable</em> command at the
+top of the stack.
 
-Undoable commands: commands that modify the WorkBook's content (`add`, `delete`, `edit` and `clear`)
 
 Examples:
-* `delete 1`
-
-    `list`
-
-    `undo` (reverses the `delete 1` command)
-
-
-* `find google`
+* `add c/Meta r/Frontend Engineer s/Application Sent d/29-Oct-2022 12:00 e/hrmonkey@example.com` followed by `undo` will
+return the WorkBook state without the added internship.
+* `edit 1 s/Behavioural Interview e/hr@meta.com` followed by `clear` and then `undo` will undo the 
+clearing of the Workbook. The edited stage and email address of the first internship will remain in the 
+returned WorkBook state.
+* `edit 2 l/golang t/` followed by `delete 1` and then 2 consecutive `undo` commands will first restore the 
+WorkBook to its state before the `delete 1` command was executed, and then to its state before the `edit` command 
+was executed.
+* `delete 2` followed by `list` and then `undo` will restore the WorkBook to its state prior to the 
+`delete 2` command as the `list` command is not an <em>undoable</em> command and does not change the 
+state of the WorkBook.
 
-  `list`
-
-  `undo` 
-
-    The `undo` command fails as there are no undoable commands executed previously.
+<div markdown="block" class="alert alert-info">
 
-    
-* `delete 1`
-    
-    `clear`
+**:information_source: Note:**<br>
+You can only undo <em>undoable</em> commands!
+* `add`
+* `edit`
+* `delete`
+* `clear`
+* `redo`
+</div>
 
-    `undo` (reverses the `clear` command)
+<div markdown="block" class="alert alert-success">
 
-    `undo` (reverses the `delete 1` command )
+**:bulb: Tip:** <br>
+* If you have undone your previous undoable command by mistake, fret not as you can easily redo this 
+to reapply the initial changes!
+* Stacking multiple `undo` commands may confuse you as to which <em>undoable</em> command you are
+undoing and which state your WorkBook is in!
```

Comment:
let's keep the user in mind again, some may not know what 'state' is (especially those control F ppl), so u can refer to the comment i made above about `undo` and try tweaking this to make it more user-friendly